atlasdb-client/src/integrationInput/java/com/palantir/atlasdb/table/description/generated/SchemaApiTestV2Table.java

```diff
@@ -0,0 +1,273 @@
+                     SchemaApiTestTable.SchemaApiTestRowResult::getColumn1));
+    }
+
+    /**
+     * Returns the value for column Column2 and specified row components. */
+    public Optional<String> getColumn2(String component1) {
+        SchemaApiTestTable.SchemaApiTestRow row = SchemaApiTestTable.SchemaApiTestRow.of(component1);
+        byte[] bytes = row.persistToBytes();
+        ColumnSelection colSelection = 
+                ColumnSelection.create(Collections.singletonList(PtBytes.toCachedBytes("d")));
+        RowResult<byte[]> rowResult = t.getRows(tableRef, ImmutableSet.of(bytes), colSelection).get(bytes);
+        if (rowResult == null) {
+            return Optional.empty();
+        }
+        else {
+            return Optional.of(SchemaApiTestTable.SchemaApiTestRowResult.of(rowResult).getColumn2());
+        }
+    }
+
+    /**
+     * Returns a mapping from row keys to value at column Column2. Ordering in the
+     * map is not guaranteed. As the values are all loaded in memory, do not use
+     * for large amounts of data. If the column does not exist for a key, the entry
+     * will be omitted from the map. */
+    public Map<String, String> getColumn2(Iterable<String> rowKeys) {
+        ColumnSelection colSelection = 
+                 ColumnSelection.create(Collections.singletonList(PtBytes.toCachedBytes("d")));
+        List<SchemaApiTestTable.SchemaApiTestRow> rows = Lists
+                .newArrayList(rowKeys)
+                .stream()
+                .map(SchemaApiTestTable.SchemaApiTestRow::of)
+                .collect(Collectors.toList());
+        SortedMap<byte[], RowResult<byte[]>> results = t.getRows(tableRef, Persistables.persistAll(rows), colSelection);
+        return results
+                .values()
+                .stream()
+                .map(entry -> SchemaApiTestTable.SchemaApiTestRowResult.of(entry))
+                .collect(Collectors.toMap(
+                     entry -> entry.getRowName().getComponent1(), 
+                     SchemaApiTestTable.SchemaApiTestRowResult::getColumn2));
+    }
+
+    /**
+     * Returns a mapping from all the row keys to their value at column Column2 (if that column exists).
+     * Ordering in the map is not guaranteed. As the values are all loaded in memory, do not use
+     * for large amounts of data.  */
+    public Map<String, String> getAllColumn2() {
+        ColumnSelection colSelection = 
+                ColumnSelection.create(Collections.singletonList(PtBytes.toCachedBytes("d")));
+        return getRowRangeColumn2(RangeRequest.all());
+    }
+
+    /**
+     * Returns a mapping from all the row keys in a rangeRequest to their value at column Column2
+     * (if that column exists). Ordering in the map is not guaranteed. As the values are all
+     * loaded in memory, do not use for large amounts of data.  */
+    public Map<String, String> getRowRangeColumn2(RangeRequest rangeRequest) {
```

Comment:
For a range scan, i think we want to return key-value pairs in order. 

Also - do we want to return a stream here instead? It feels a little dangerous to return a map. I do like the ability to be able to easily turn it into a map (maybe even a `LinkedHashMap` that preserves ordering), but might be best to have a `getSmallColumn1Range` method or a `toMap()` utility method on the returned stream .

Reply:
Changed it so that it uses a LinkedHashMap and ordering is preserved. 
Returning a Stream of RowResults at this stage would be equally bad as immutableCopy() loads everything into a list anyway. Have renamed to "getSmallRowRange" for now, and support for Streams can be added once the Transaction class has a method that exposes the results as a smart Stream, instead of a BatchingVisitable. What do you think?